### PR TITLE
Some working methods

### PR DESCRIPTION
**Pull request by kaumudpa**
getBalance is working in example
we have also added a getTokenBalance button in example which is getBalance method only but fetches balance for token.

Also readContract is working now - check `Get Token Supply From Contract Button` in example.

Contracts to test deployes on Testnet - Polygon Amoy 

Do let us know your comments 
@Chralu & @redDwarf03

### CODE REVIEW

**Comment by kaumudpa** [comment]
@redDwarf03 Found the issue which you might be facing.

`required this.decimal,` should be required `this.decimals,`

we need to use `decimals` instead of `decimal` across the project.

**Comment by redDwarf03** [comment]
> @redDwarf03 Found the issue which you might be facing.
> 
> `required this.decimal,` should be required `this.decimals,`
> 
> we need to use `decimals` instead of `decimal` across the project.

I spent an incredible amount of time finding this problem 😱

File: lib/src/js/actions/read_contract.js.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
+part of '../wagmi.js.dart';
+
+/// [Documentation API](https://wagmi.sh/core/api/actions/getToken)
```
Comment (redDwarf03) [inline lib/src/js/actions/read_contract.js.dart:3]: https://wagmi.sh/core/api/actions/readContract

File: lib/src/js/actions/read_contract.js.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+@JS()
+extension type JSReadContractParameters._(JSObject _) implements JSObject {
+  external JSReadContractParameters({
+    required JSAny? abi,
```
Comment (redDwarf03) [inline lib/src/js/actions/read_contract.js.dart:7]: Why nullable with ? if required

**Comment by redDwarf03** [comment]
Thank you for your work
I hope you're happy with the lib for now. It lacks error handling, which we'll be working on next week.

**Comment by kaumudpa** [comment]
@redDwarf03 Lots of critical actions yet to be implementations - Lets work on implementing - 

`call, estimateFeesPerGas, estimateGas, estimateMaxPriorityFeePerGas, getTransactionCount, getTransaction, getTransactionConfirmations,  getTransactionReceipt, multicall, sendTransaction, switchAccount, waitForTransactionReceipt, writeContract `

and then move towards proper error handling - we will contribute as well.

**Comment by redDwarf03** [comment]
@kaumudpa we can help each other here about technical flutter questions: https://github.com/archethic-foundation/wagmi_flutter_web/discussions/3

**Pull request merged**